File: JavaScript/Swap alternate.js
function swapAlternate(arr, size) {
  for (let i = 0; i < size; i += 2) {
    if (i + 1 < size) {
      // Swap using destructuring assignment
      [arr[i], arr[i + 1]] = [arr[i + 1], arr[i]];
    }
  }
}

// Using readline for user input
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

console.log("Enter array elements (10 numbers): ");

let numsArr = [];
let count = 0;

rl.on("line", (input) => {
  numsArr.push(parseInt(input));
  count++;

  if (count === 10) {
    console.log("\nBefore swap:");
    console.log(numsArr.join("\n"));

    swapAlternate(numsArr, 10);

    console.log("\nAfter swap:");
    console.log(numsArr.join("\n"));

    rl.close();
  }
});

File: Python/Swap alternate.py
def swap_alternate(arr):
    for i in range(0, len(arr), 2):
        if (i + 1) < len(arr):
            arr[i], arr[i + 1] = arr[i + 1], arr[i]


# Taking user input
print("Enter array elements (10 numbers): ")
nums_arr = []

for i in range(10):
    nums_arr.append(int(input()))

print("\nBefore swap:")
for num in nums_arr:
    print(num)

swap_alternate(nums_arr)

print("\nAfter swap:")
for num in nums_arr:
    print(num)
